runtime/src/qio/qio_popen.c

```diff
@@ -447,7 +454,14 @@ qioerr qio_proc_communicate(
   while( do_input || do_output || do_error ) {
     if( do_input ) {
       err = _qio_channel_flush_qio_unlocked(input);
-      if( !err ) do_input = false;
+      if( !err ) {
+        qioerr err2 = 0;
+        do_input = false;
+        // Close input channel.
+        err = qio_channel_close(false, input);
+        err2 = qio_file_close(input_file);
+        if( err2 && !err ) err = err2;
```

Comment:
So we only care about an error on file_close if there wasn't an error closing the channel?  Is it reasonable to still try the file_close if there was a channel_close error?